inference-engine/tests/functional/plugin/myriad/single_layer_tests/proposal.cpp

```diff
@@ -0,0 +1,427 @@
+*/
+
+static float check_iou(const float* A, const float* B) {
+    if (A[0] > B[2] || A[1] > B[3] || A[2] < B[0] || A[3] < B[1]) {
+        return 0.0f;
+    } else {
+        // overlapped region (= box)
+        const float x1 = std::max(A[0], B[0]);
+        const float y1 = std::max(A[1], B[1]);
+        const float x2 = std::min(A[2], B[2]);
+        const float y2 = std::min(A[3], B[3]);
+
+        // intersection area
+        const float width = std::max(0.0f, x2 - x1 + 1.0f);
+        const float height = std::max(0.0f, y2 - y1 + 1.0f);
+        const float area = width * height;
+
+        // area of A, B
+        const float A_area = (A[2] - A[0] + 1.0f) * (A[3] - A[1] + 1.0f);
+        const float B_area = (B[2] - B[0] + 1.0f) * (B[3] - B[1] + 1.0f);
+
+        // IoU
+        return area / (A_area + B_area - area);
+    }
+}
+
+static float check_iou_normalized(const float* A, const float* B) {
+    if (A[0] > B[2] || A[1] > B[3] || A[2] < B[0] || A[3] < B[1]) {
+        return 0.0f;
+    } else {
+        // overlapped region (= box)
+        const float x1 = std::max(A[0], B[0]);
+        const float y1 = std::max(A[1], B[1]);
+        const float x2 = std::min(A[2], B[2]);
+        const float y2 = std::min(A[3], B[3]);
+
+        // intersection area
+        const float width = std::max(0.0f, x2 - x1);
+        const float height = std::max(0.0f, y2 - y1);
+        const float area = width * height;
+
+        // area of A, B
+        const float A_area = (A[2] - A[0]) * (A[3] - A[1]);
+        const float B_area = (B[2] - B[0]) * (B[3] - B[1]);
+
+        // IoU
+        return area / (A_area + B_area - area);
+    }
+}
+
+static std::size_t get_num_rois(const float* array, std::size_t size) {
+    std::size_t count = 0;
+    while (count < size && array[count] != -1.f)
+        count += 5;
+    return count / 5;
+}
+
+typedef std::vector<std::vector<int>> Graph;
+
+class GraphComparison {
+public:
+    int calculateMatchingCount(const float* reference, std::size_t count_reference,
+                               const float* optimized, std::size_t count_optimized,
+                               const float* reference_scores, const float* output_scores,
+                               bool normalized, bool withOutputScores) {
+        validateData(reference, count_reference, optimized, count_optimized);
+
+        //edges from vertexes of optimized boxes to vertexes of reference boxes
+        Graph graph {make_graph(reference, count_reference, optimized, count_optimized,
+                                reference_scores, output_scores,
+                                normalized, withOutputScores)};
+
+        int matching_count = 0;
+        std::vector<bool> used(count_optimized, false);
+        std::vector<int> mt(count_reference, -1); // matching, -1 means there is no matching
+        for (std::size_t vertex = 0; vertex < count_optimized; ++vertex) {
+            if (try_kuhn(used, mt, graph, vertex)) {
```

Comment:
Interface looks a bit strange: `try_kuhn` takes `mt` as an argument, but this is not changed anywhere outside of function - looks like `mt` is some internal container for algorithm. If so it is better to encapsulate it inside class.

Reply:
fixed